libs/data-mapper-v2/src/utils/ReactFlow.Util.ts

```diff
@@ -82,29 +83,28 @@ export const convertWholeDataMapToLayoutTree = (
   let nextEdgeIndex = 0;
   const layoutEdges: LayoutEdge[] = [];
 
+  // loops through all connections and adds if not a custom value
   Object.values(connections).forEach((connection) => {
-    Object.values(connection.inputs).forEach((inputValueArray, inputIndex) => {
-      inputValueArray.forEach((inputValue, inputValueIndex) => {
-        if (isConnectionUnit(inputValue)) {
-          const targetId = connection.self.reactFlowKey;
-          const labels = isFunctionData(connection.self.node)
-            ? connection.self.node?.maxNumberOfInputs > -1
-              ? [connection.self.node.inputs[inputIndex].name]
-              : [generateInputHandleId(connection.self.node.inputs[inputIndex].name, inputValueIndex)]
-            : [];
-
-          const nextEdge: LayoutEdge = {
-            id: `e${nextEdgeIndex}`,
-            sourceId: inputValue.reactFlowKey,
-            targetId,
-            labels,
-            isRepeating: inputValue.isRepeating ?? false,
-          };
-
-          layoutEdges.push(nextEdge);
-          nextEdgeIndex += 1;
-        }
-      });
+    connection.inputs.forEach((input, inputIndex) => {
+      if (isConnectionUnit(input)) {
+        const targetId = connection.self.reactFlowKey;
+        const labels = isFunctionData(connection.self.node)
+          ? connection.self.node?.maxNumberOfInputs > UnboundedInput
+            ? [connection.self.node.inputs[inputIndex].name]
+            : [generateInputHandleId(connection.self.node.inputs[0].name, inputIndex)] // if unlimited inputs, the name is always the same
+          : [];
```

Comment:
Some more comment on what you are actually doing here will help when a new person reads this code

Reply:
I extracted some of this, I think one problem is that the nested ternary operators are confusing

Reply:
And renamed isConnectionUnit to isNodeConnection